zipline/pipeline/factors/factor.py

```diff
@@ -621,6 +621,193 @@ def rank(self, method='ordinal', ascending=True, mask=NotSpecified):
         """
         return Rank(self, method=method, ascending=ascending, mask=mask)
 
+    @expect_types(
+        target=Slice, correlation_length=int, mask=(Filter, NotSpecifiedType),
```

Comment:
Yup, that's already being done here: https://github.com/quantopian/zipline/pull/1291